compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEFoldDmaWaits.cpp

```diff
@@ -16,13 +16,58 @@ namespace mlir::iree_compiler::AMDAIE {
 
 namespace {
 
-/// Utility function to determine whether a DMA wait op can be folded based on
-/// its half DMA copy operation.
-FailureOr<bool> canFoldBasedOnHalfDmaCpy(
-    const AMDAIE::AMDAIEDeviceModel &deviceModel,
+using DmaBdIdKey = std::pair<AMDAIE::TileOp, AMDAIE::ConnectionOp>;
+
+/// Utility function to erase the DMA wait operations in the queue, except for
+/// the last one.
+LogicalResult eraseQueueOperations(IRRewriter &rewriter,
```

Comment:
I think this is more of a refactoring change and isn't specifically needed for DMA folding across multiple columns? If so, try to avoid combining multiple changes in one PR as it makes reviewing harder and it makes it unclear where these changes come from if someone needs to look at this in the future.